metpy/io/pandas_to_netcdf.py

```diff
@@ -0,0 +1,166 @@
+        individual points
+
+        Parameters
+        ----------
+        df : `pandas.DataFrame`
+            Point data in pandas dataframe.
+
+        sampling_var : str
+            Column name that is the sampling dimension: for surface observations,
+            this is the column that contains the station identifier/name
+
+        sampling_data_vars : list
+            List of all variables associated with the sampling variable that do not
+            vary with time, such as latitude, longitude, and elevation for
+            surface observations
+
+        path_to_save : str
+            Path, including filename, for where to save netCDF file.
+
+        netcdf_format : str, optional
+
+        column_units : dict, optional
+            Dictionary of units to attach to columns of the dataframe. Overrides
+            the units attribute if it is attached to the dataframe.
+
+        standard_names : dict, optional
+            Dictionary of variable descriptions that are CF-compliant
+
+        long_names : dict, optional
+            Dictionary of longer variable descriptions that provide more detail
+            than standard_names
+
+        dataset_type: str, optional
+            Type of dataset to be converted. Options are 'timeSeries', 'profile',
+            or 'trajectory'. While optional, this variable should be declared to create
+            a CF-compliant DSG netCDF file.
+
+        Returns
+        -------
+        NetCDF file saved to `path_to_save`.
+
+        """
+        # Verify_integrity must be true in order for conversion to netCDF to work
+        # Return a TypeError if not provided a Pandas DataFrame
+        try:
+            # Create the dimensions for use later in netCDF file
+            samplingindex = df.groupby([sampling_var], sort=False).ngroup()
+            obs = arange(0, len(df))
+            df.insert(0, 'samplingIndex', samplingindex)
+            df.insert(1, 'observations', obs)
+
+            # Handle the sampling location specific data
+            sampling_data = df[sampling_data_vars]
+            samples = sampling_data.groupby([sampling_var], sort=False).ngroup()
+            sampling_data.insert(0, 'samples', samples)
+            sampling_data = sampling_data.groupby('samples').first()
+            dataset_samples = xr.Dataset.from_dataframe(sampling_data)
+
+            # Create the dataset for the variables of each observation
+            df = df.drop(sampling_data_vars, axis=1)
+            df = df.set_index(['observations'], verify_integrity=True)
+            dataset_var = xr.Dataset.from_dataframe(df)
+
+            # Merge the two datasets together
+            dataset_final = xr.merge([dataset_samples, dataset_var], compat='no_conflicts')
+
+        except (AttributeError, ValueError, TypeError):
+            raise TypeError('A pandas dataframe was not provided')
+
+        # Attach variable-specific metadata
+        if column_units:
+            column_units['samples'] = ''
```

Comment:
Since the user is passing in a dict, it's not good for a library to modify that in-place. Also, do you intend to override if the user passes in units for e.g. 'samples'?

One pattern would be:
```python
final_column_units = {}
final_column_units['samples'] = ''
final_column_units.update(column_units)
```
assuming you want all user values to take precedence.

Reply:
Just getting back around to this. Yeah this was an oversight on my part - mainly due to the fact that these variables are created within the function, so I assumed that the user wouldn't provide anything for them, but that's a poor assumption, and something that should probably be added in the docstring to include those in the dictionary if desired.